programmieren-lernen/dependency-injection.md

```diff
@@ -6,26 +6,39 @@ alias: ["di"]
 Dependency Injection by Example:
 ```java
 public class MeineApp {
+  // Hier übergibt man die Instanz der MeineApp Klasse
   private final DepA a = new DepA(this);
-  private final Depb b = new DepB(a);
+  // Die DepB Klasse greift auf die DepA Klasse zu
+  public final DepB b = new DepB(a);
+  // Info: Es muss nicht unbedingt bei der Deklaration passieren,
+  // im spigot Kontext, ist es auch in der onEnable möglich,
+  // es geht darum beim Konstruktor aufruf zu übergeben.
   public void start(String[] args) {
-    // Do Stuff
+    b.doThat();
   }
-  // Getters here
 }
 ```
 ```java
 public class DepA {
   private final MeineApp app;
+  // Hier kann man dann die Instanz der MeineApp Klasse verwenden.
   public DepA(MeineApp app) { this.app = app; }
-  public void doThis() {}
+  // Um in dem Fall um auf b Zugreifen, man muss sich aber bemühen nur die Sachen zu übergeben,
+  // die auch verwendet werden, hier also eher eine DepB Instanz,
+  // aber da diese nach DepA initialisiert wird, kann man die nicht direkt übergeben.
```

Comment:
Bitte in ein inhaltlich und grammatikalisch richtiges Deutsch bringen.

Reply:
If you have some problems with my German, provide suggestions instead. This tag existed for years now and nobody cared.